Fix/lapi empty response

Nydauron opened the pull request:
- Fixes the default config to work with the default LAPI config.
- Adds extra error handling to properly distinguish when the LAPI request 404s due to double backslashes (e.g. `http://localhost:8080//v1/decisions?ip=`). Before this patch, `crowdsec_proxy()` would return `OK` with the response being the empty string.

Nydauron committed:
Changes default config LAPI URL to prevent 404ing

This is more or less a bandage fix and doesn't fix the underlying
problem within the module.

Nydauron committed:
Fixes double slash URLs to properly report back a 404 error

blotus commented:
Hey,

Thanks for the PR !

What do you think about also removing automatically extra `/` at the end of the provided URL ?

That way, the bouncer will work in both case, and it limits the risk of user error.

Nydauron commented:
> What do you think about also removing automatically extra `/` at the end of the provided URL ?
> 
> That way, the bouncer will work in both case, and it limits the risk of user error.

Sure, I think that would be a good idea. It would make sense if we simply stored the schema and authority prefix without any trailing slashes inside `sconf->url` when the module sets it with `CrowdSecURL` (e.g. `http://127.0.0.1`).

Some other things to think are also:
- Would we want to add a parsing check for the URL to prevent apache from starting up with a bad URL?
  - An error message can then be provided that gets printed to logs
- Is `CrowdSecURL` strictly expecting a `<schema>://<authority>/` format, or would we allow a say values without the schema (e.g. `<hostname>:<port>`) to also be valid?

Nydauron committed:
Changes conditions to use apache OK status constant

Nydauron committed:
Adds validation when Crowdsec LAPI URL is set

This simply parses the URL string to ensure the prefix contains a scheme
and an authority, following the RFC 3986 standard. If the scheme and
authority cannot be parsed due to the passed string not following the
RFC standard, Apache will fail to start and print an error to logs.

The path, query, and fragment are subsequently ignored and a warning
gets printed if anything besides the string "/" is found after the
authority.

Nydauron commented:
Went ahead and implemented a URL validation check when `CrowdsecURL` gets set. Since most of the other bouncers expect at least a schema and authority in the URL, it makes sense to check if the URL prefix follows `<scheme>://<authority>` and ignores the rest if there is any.
